src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -1417,6 +1417,34 @@ QgsCoordinateReferenceSystem QgsProcessingParameters::parameterAsPointCrs( const
     return QgsCoordinateReferenceSystem();
 }
 
+QgsGeometry QgsProcessingParameters::parameterAsGeometry( const QgsProcessingParameterDefinition *definition, const QVariantMap &parameters, QgsProcessingContext &context )
+{
+  if ( !definition )
+    return QgsGeometry();
+
+  return parameterAsGeometry( definition, parameters.value( definition->name() ), context );
+}
+
+QgsGeometry QgsProcessingParameters::parameterAsGeometry( const QgsProcessingParameterDefinition *definition, const QVariant &value, QgsProcessingContext &context )
+{
+  if ( !definition )
+    return QgsGeometry();
+
+  QVariant val = value;
+  if ( val.canConvert< QgsGeometry >() )
+  {
+    return val.value<QgsGeometry>();
+  }
+
```

Comment:
I think this method should also check for QgsRectangle or QgsPointXY values -- those are acceptable for the extent and point parameter types respectively, and since the name of this method is generic then it would be good to be as forgiving/tolerant as possible here

Reply:
I hesitated because there was already  the point and extent parameter, but that's makes perfect sense to me.

Reply:
Done

Added supports for QgsReferencedRectangle and QgsReferencedPointXY.